pkg/aws/aws.go

```diff
@@ -98,26 +102,30 @@ func New(config *Config) (*AWS, error) {
 	for _, service := range services {
 		switch service {
 		case "S3":
-			// There are two scenarios:
-			// 1. Running locally, the user must pass in a region and profile to use
-			// 2. Running within an EC2 instance and the region and profile can be derived
-			// I'm going to use the AWS SDK to handle this for me. If the user has provided a region and profile, it will use that.
-			// If not, it will use the EC2 instance metadata service to determine the region and credentials.
-			// This is the same logic that the AWS CLI uses, so it should be fine.
-			options := []func(*awsconfig.LoadOptions) error{awsconfig.WithEC2IMDSRegion()}
-			if config.Region != "" {
-				options = append(options, awsconfig.WithRegion(config.Region))
-			}
-			if config.Profile != "" {
-				options = append(options, awsconfig.WithSharedConfigProfile(config.Profile))
-			}
-			ac, err := awsconfig.LoadDefaultConfig(context.Background(), options...)
-			if err != nil {
-				return nil, err
+			var clients []costexplorer.CostExplorer
+			for _, profile := range config.Profiles {
+				// There are two scenarios:
+				// 1. Running locally, the user must pass in a region and profile to use
+				// 2. Running within an EC2 instance and the region and profile can be derived
+				// I'm going to use the AWS SDK to handle this for me. If the user has provided a region and profile, it will use that.
+				// If not, it will use the EC2 instance metadata service to determine the region and credentials.
+				// This is the same logic that the AWS CLI uses, so it should be fine.
+				options := []func(*awsconfig.LoadOptions) error{awsconfig.WithEC2IMDSRegion()}
+				if config.Region != "" {
+					options = append(options, awsconfig.WithRegion(config.Region))
+				}
+				if config.Profile != "" {
+					options = append(options, awsconfig.WithSharedConfigProfile(profile))
+				}
```

Comment:
I'm not sure what's going on here. You check here that the single `-profile` flag has some value, and if it does, then forget about that value and use the one from the iteration 🤔 

Basically the `-profile` value never gets used, and in fact, I'd argue than adding this flag adds confusion to the user, as now they have `-profile` and `-profiles`. It seems to me that the single case is the same as `-profiles` with just one value.

What am I missing?

Reply:
Pretty on point question. Mainly, naming is hard 😅 

It looks like we're not using the `-profile` flag in dev or production today:

```
 spec:
  ...
      containers:
        - args:
            - --provider=aws
            - --aws.services=s3
```

I think the initial intent of the check you see is when doing local development to figure out which profile we should target. In dev/production that should be handled by:

```go
options := []func(*awsconfig.LoadOptions) error{awsconfig.WithEC2IMDSRegion()}
```

Let me remove the `-profile` flag as it's not used anymore. I'll build a local image and deploy this into development and make sure it still works 😬

Reply:
agreed here, it looks like this might've been a misstep.  It now appears that we can't even use the flags separately, they _must_ be used together, as specifying `-aws.profile` will not get used unless there's a profile from `-aws.profiles` as well, in which case it'll use the value(s) from `aws.profiles`, but _unless_ you supply `aws.profile`, it'll never use any config values 😕

Reply:
```suggestion
				options = append(options, awsconfig.WithSharedConfigProfile(profile))
```

I'd suggest something like this, perhaps with a conditional flag to look if `aws-profile` is defined first rather than a loop through Profiles?

Or better yet, I'd just eliminate the `aws.profiles` flag entirely and allow `aws.profile` to be specified multiple times OR eliminate `aws.profile` and allow `aws.profiles` to be one or more comma-seperated items.

Reply:
I've opted to stick with `-aws.profiles` as a flag, as I think it makes it clear that multiple values can be passed in. @inkel @logyball I've pushed a commit that removes the reference, and appreciate the thoughts/suggestions!

`cloudcost-exporter` when deployed infers the profile and region from this line:

```golang
				options := []func(*awsconfig.LoadOptions) error{awsconfig.WithEC2IMDSRegion()}
```

I need to be a bit mindful when rolling this change out. If the `-profiles` isn't set, I've added logic to return an error. So I'll need to likely couple rolling out the wave with adding the `-aws.profiles` flag to the environment.